### PR TITLE
go/store/nbs: table_index.go: Fix uint32 overflow in findPrefix when working with large table files.

### PR DESCRIPTION
**Pull request by reltuk**

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@reltuk __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>07659b3</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>07659b3</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@reltuk __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>5c7324e</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>5c7324e</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@reltuk __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>9f3ab6b</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>9f3ab6b</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: go/store/nbs/table_index_test.go [hunk 1]
```diff
@@ -147,6 +147,42 @@ func BenchmarkFindPrefix(b *testing.B) {
 	})
 }
 
+func TestFindPrefix(t *testing.T) {
+	if isRaceEnabled() {
+		t.Skip("We do not run TestFindPrefix in race mode because of its large memory requirement")
+	}
+	if os.Getenv("CI") != "" {
```
Comment (macneale4) [inline go/store/nbs/table_index_test.go:154]: This is a thing? I didn't know the CI env var this existed.

File: go/store/nbs/table_index_test.go [hunk 2]
```diff
@@ -147,6 +147,42 @@ func BenchmarkFindPrefix(b *testing.B) {
...
+		t.Skip("We do not run TestFindPrefix in CI for now because of its large memory requirement")
+	}
+
+	// Test some edge cases in findPrefix.
+	var idx onHeapTableIndex
+	idx.count = 1
+	idx.prefixTuples = make([]byte, 12)
+	assert.Equal(t, uint32(0), idx.findPrefix(0))
+	assert.Equal(t, uint32(1), idx.findPrefix(1))
```
Comment (macneale4) [inline go/store/nbs/table_index_test.go:163]: I really don't like tests where there is a const of 1 everywhere. I'll push a suggestion to this PR.

**Review by macneale4 — APPROVED** [review]
I approved, but also added another test. Ship it!

**Comment by coffeegoddd** [comment]
@macneale4 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>c2c1007</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>c2c1007</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>
